src/OpenTelemetry.Extensions.PersistentStorage.Api/IPersistentBlob.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="IPersistentBlob.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Api
+{
+    /// <summary>
+    /// Represents a persistent blob.
+    /// </summary>
+    public interface IPersistentBlob
+    {
+        /// <summary>
+        /// Reads the content from the blob.
+        /// </summary>
+        /// <returns>
+        /// The content of the blob if the operation succeeded, otherwise null.
+        /// </returns>
+        /// <remarks>
+        /// This function should never throw exception.
+        /// </remarks>
+        byte[] Read();
+
+        /// <summary>
+        /// Writes the given content to the blob.
+        /// </summary>
+        /// <param name="buffer">
+        /// The content to be written.
+        /// </param>
+        /// <param name="leasePeriodMilliseconds">
+        /// The number of milliseconds to lease after the write operation finished.
+        /// </param>
+        /// <returns>
+        /// The same blob if the operation succeeded, otherwise null.
+        /// </returns>
+        /// <remarks>
+        /// This function should never throw exception.
+        /// </remarks>
+        IPersistentBlob Write(byte[] buffer, int leasePeriodMilliseconds = 0);
```

Comment:
Just reading the comments about the behavior, maybe `bool TryWrite(byte[] buffer, int leasePeriodMilliseconds = 0)` might be a better signature. For the others in this interface...

```csharp
bool TryRead(out byte[] buffer);
bool TryLease(int leasePeriodMilliseconds);
bool TryDelete();
```
They seem to all "attempt" some action. I think "Try" is a good pattern for that behavior. Could just be the comments are misleading 😄

Reply:
Just a little bit more on the pattern, you can offer both signatures if you want like...

```csharp
public void Write(byte[] buffer, int leasePeriodMilliseconds = 0)
{
    if (!TryWrite(buffer, leasePeriodMilliseconds))
       throw new InvalidOperationException(...);
}

public bool TryWrite(byte[] buffer, int leasePeriodMilliseconds = 0)
{
    return AttemptWrite(buffer, leasePeriodMilliseconds);
}
```

Reply:
+1 to returning a boolean success rather than `IPersistentBlob` assuming its only purpose was to communicate success.